theseus/geometry/se2.py

```diff
@@ -23,14 +23,14 @@ def __init__(
         data: Optional[torch.Tensor] = None,
         name: Optional[str] = None,
         dtype: Optional[torch.dtype] = None,
-        requires_check: bool = True,
+        strict: bool = False,
     ):
         if x_y_theta is not None and data is not None:
             raise ValueError("Please provide only one of x_y_theta or data.")
         if x_y_theta is not None:
             dtype = x_y_theta.dtype
-        if data is not None and requires_check:
-            self._data_check(data)
+        if data is not None:
```

Comment:
Just realized that we can move this check (which happens in all subclasses) to `Manifold`'s constructor (right after line 40), so that the main logic is contained in one place, making it easier to maintain in the future. Sorry for missing this before!